### PR TITLE
Milestone 5: Friendships v1

### PR DESCRIPTION
**Pull request by bolah2009**
## Milestone 5: Friendships v1

Create a model with associations and all requested features for friendships.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -17,4 +19,24 @@ class User < ApplicationRecord
   def liked?(post_id)
     likes_given.where('post_id = ?', post_id).any?
   end
+
+  def pending_requests
+    Friendship.where('friend_id = ? AND confirmed = ?', id, false)
+  end
```
Comment (mosaaleb) [inline app/models/user.rb]: - [ ] **`[OPTIONAL]`** We can simplify this with 

`Friendship.where(confirmed: false)` 

Same for `sent_requests`

File: app/models/friendship.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class Friendship < ApplicationRecord
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
+
+  validates :user_id, presence: true
+  validates :friend_id, presence: true
+  validate :friend_cannot_be_user
```
[the remark this answers is not in the document]
Comment (bolah2009) [reply app/models/friendship.rb]: Thanks for the review. Will do :+1:

File: app/models/user.rb (hunk 1 above)
Comment (bolah2009) [reply app/models/user.rb]: Hey @mosaaleb, this suggestion is not accurate. :thinking: 

Using `Friendship.where('friend_id = ? AND confirmed = ?', id, false)` checks the Friendship table where `friend_id` is the user id. Omitting `friend_id` will return data for both sent and pending request. 

Kindly, review this and get back to us. 

Thanks.

Comment (mosaaleb) [reply app/models/user.rb]: You are right :+1: Sorry about this inconvenience.

**Comment by bolah2009** [comment]
Hi @mcrd25, thanks for the review :ok_hand: 

We are closing this PR and making the suggested changes in this PR #7 to avoid conflict as there's another review ongoing there. Thanks. :+1:

File: config/routes.rb [hunk 3]
```diff
@@ -12,5 +12,5 @@
 
   resources :likes
 
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :friendships
```
[the remark this answers is not in the document]
Comment (bolah2009) [reply config/routes.rb]: Okay :+1:

**Pull request closed without merging**